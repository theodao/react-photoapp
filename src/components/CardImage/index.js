import React from 'react';
import styles from './styles.module.scss';

export default ({ src, onClick = () => {} }) => {
  return (
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
    <div
      className={styles.card}
      onClick={() => onClick()}
      role="button"
      tabIndex="0"
    >
      <div className={styles.cardImage}>
        {' '}
        <img src={src} alt="#placeholder" />{' '}
      </div>
      <div className={styles.cardTitle}>Category</div>
    </div>
  );
};
