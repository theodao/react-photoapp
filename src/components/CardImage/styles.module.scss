.card {
  margin: 30px auto;
  width: 100%;
  height: 100%;
  transition: 0.4s;
  border-radius: 40px;
  .cardImage {
    width: inherit;
    height: inherit;
    border-radius: 40px;
    img {
      width: inherit;
      height: inherit;
      border-radius: 40px;
      object-fit: cover;
    }
  }

  .cardTitle {
    text-align: center;
    border-radius: 0px 0px 40px 40px;
    font-family: sans-serif;
    font-weight: bold;
    font-size: 25px;
    color: white;
    margin-top: -45px;
    height: 40px;
  }

  &:hover {
    transform: scale(0.9, 0.9);
    box-shadow: 5px 5px 30px 15px rgba(0, 0, 0, 0.25),
      -5px -5px 30px 15px rgba(0, 0, 0, 0.22);
    transition: 0.4s;
  }
}
