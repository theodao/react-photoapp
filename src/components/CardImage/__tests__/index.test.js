import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { shallow } from 'enzyme';
import CardImage from '../';

describe('Testing card image component', () => {
  afterEach(cleanup);

  it('Should match snapshot', () => {
    const { asFragment } = render(<CardImage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should trigger on click', () => {
    const onClickButton = jest.fn();

    const wrapper = shallow(<CardImage onClick={onClickButton} />);

    wrapper.find('.card').props().onClick();

    expect(onClickButton).toHaveBeenCalled();
  });
});
