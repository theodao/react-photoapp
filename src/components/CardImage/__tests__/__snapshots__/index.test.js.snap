// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing card image component Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
    role="button"
    tabindex="0"
  >
    <div
      class="cardImage"
    >
       
      <img
        alt="#placeholder"
      />
       
    </div>
    <div
      class="cardTitle"
    >
      Category
    </div>
  </div>
</DocumentFragment>
`;
